code/modules/mob/living/carbon/human/life.dm

```diff
@@ -74,15 +74,20 @@
 
 
 /mob/living/carbon/human/calculate_affecting_pressure(pressure)
-	if (wear_suit && head && istype(wear_suit, /obj/item/clothing) && istype(head, /obj/item/clothing))
-		var/obj/item/clothing/CS = wear_suit
-		var/obj/item/clothing/CH = head
-		if (CS.clothing_flags & CH.clothing_flags & STOPSPRESSUREDAMAGE)
-			return ONE_ATMOSPHERE
-	else if(!get_bodypart(BODY_ZONE_HEAD) && wear_suit && istype(wear_suit, /obj/item/clothing)) // you don't need a helmet if you don't have a head
-		var/obj/item/clothing/CS = wear_suit
-		if(CS.clothing_flags & STOPSPRESSUREDAMAGE)
-			return ONE_ATMOSPHERE
+	var/obj/item/clothing/CS = wear_suit
+	if(CS && istype(CS))
+		if(get_bodypart(BODY_ZONE_HEAD))
+			var/obj/item/clothing/CH = head
+			if(CH && istype(CH))
+				if(pressure > ONE_ATMOSPHERE && (CS.clothing_flags & CH.clothing_flags & STOPSHIGHPRESSURE))
+					return ONE_ATMOSPHERE
+				else if(pressure < ONE_ATMOSPHERE && (CS.clothing_flags & CH.clothing_flags & STOPSLOWPRESSURE))
+					return ONE_ATMOSPHERE
+		else // you don't need a helmet if you don't have a head
+			if(pressure > ONE_ATMOSPHERE && (CS.clothing_flags & STOPSHIGHPRESSURE))
+				return ONE_ATMOSPHERE
+			else if(pressure < ONE_ATMOSPHERE && (CS.clothing_flags & STOPSLOWPRESSURE))
+				return ONE_ATMOSPHERE
```

Comment:
mfw else if after return...

Reply:
realistically, we shouldn't be having several return statements if we were following best coding practices, but training is fighting with shitcode